app/code/Magento/InventoryShipping/Ui/DataProvider/SourceSelectionDataProvider.php

```diff
@@ -55,6 +62,11 @@ class SourceSelectionDataProvider extends AbstractDataProvider
      */
     private $sourceRepository;
 
+    /**
+     * @var GetStockItemConfigurationInterface
+     */
+    private $getStockItemConfiguration;
```

Comment:
This class currently contains 13 external dependencies, which leads to static tests failure.

If you will move next logic:

```php
    /**
     * @param $stockId
     * @param $sku
     * @param $qty
     * @return array
     * @throws \Magento\Framework\Exception\NoSuchEntityException
     */
    protected function getSources($stockId, $sku, $qty)
    {
        $algorithmCode = $this->getDefaultSourceSelectionAlgorithmCode->execute();

        $requestItem = $this->itemRequestFactory->create([
            'sku' => $sku,
            'qty' => $qty
        ]);
        $inventoryRequest = $this->inventoryRequestFactory->create([
            'stockId' => $stockId,
            'items' => [$requestItem]
        ]);
        $sourceSelectionResult = $this->sourceSelectionService->execute(
            $inventoryRequest,
            $algorithmCode
        );
        $result = [];
        foreach ($sourceSelectionResult->getSourceSelectionItems() as $item) {
            $sourceCode = $item->getSourceCode();
            $result[] = [
                'sourceName' => $this->getSourceName($sourceCode),
                'sourceCode' => $sourceCode,
                'qtyAvailable' => $item->getQtyAvailable(),
                'qtyToDeduct' => $item->getQtyToDeduct()
            ];
        }

        return $result;
    }

    /**
     * Get source name by code
     *
     * @param string $sourceCode
     * @return mixed
     * @throws \Magento\Framework\Exception\NoSuchEntityException
     */
    public function getSourceName(string $sourceCode): string
    {
        if (!isset($this->sources[$sourceCode])) {
            $this->sources[$sourceCode] = $this->sourceRepository->get($sourceCode)->getName();
        }

        return $this->sources[$sourceCode];
    }
```
into the dedicated class/service you will get rid of external dependencies on 
```
$this->getDefaultSourceSelectionAlgorithmCode
$this->itemRequestFactory
$this->inventoryRequestFactory
$this->sourceSelectionService
$this->sourceRepository
```
and test will pass